src/components/tools/text-style-bar.tsx

```diff
@@ -0,0 +1,108 @@
+import * as React from "react";
+import { inject, observer } from "mobx-react";
+import { BaseComponent } from "../base";
+import { isMac } from "../../utilities/browser";
+
+import "./text-style-bar.sass";
+
+interface IButtonDef {
+  iconName: string;  // Font-Awesome icon name for this button.
+  toolTip: string;   // Text for the button's tool-tip.
+}
+
+interface IProps {
+  selectedButtonNames: string[];
+  clickHandler: (buttonName: string, editor: any, event: React.MouseEvent) => void;
+  editor: any;
+  visible: boolean;  // If true, render the tool bar.
+  enabled: boolean;  // If true, let user events be processed.
+}
+
+// This component renders HTML for a vertical tool-bar used to style text in
+// a text-tool-editor. What follows is a pseudo-markup-outline for the various
+// HTML-tags & CSS-class names.
+//
+// <div text-style-bar>
+//   <div bar-header-icon>
+//     <svg header-icon />
+//   </div>
+//   <div button-with-tool-tip >  // This div is repeated for each button.
+//     <i button-icon fa fa-fw (on | off) [fa-bold, fa-italic, ..., fa-undo] />
+//     <span tool-tip-text />
+//   </div>
+//   ...
+// </div>
+
+@inject("stores")
+@observer
+export class TextStyleBarComponent extends BaseComponent<IProps, {}> {
+
+  private prefix = isMac() ? "Cmd-" : "Ctrl-";
+
+  // If there should be a need to use this component in more than one place,
+  // it should be relatively straightforward to abstract the set of buttons
+  // outside of the component and define the contents of the bar in the client
+  // code that instantiates the TextStyleBarComponent.
+
+  private buttonDefs: IButtonDef[] = [
+    { iconName: "bold",        toolTip: `Bold (${this.prefix}b)`},
+    { iconName: "italic",      toolTip: `Italic (${this.prefix}i)`},
+    { iconName: "underline",   toolTip: `Underline (${this.prefix}u)`},
+    { iconName: "code",        toolTip: `Typewriter Font`},
+    { iconName: "subscript",   toolTip: `Subscript (${this.prefix},)`},
+    { iconName: "superscript", toolTip: `Superscript (${this.prefix}Shift-,)`},
+    { iconName: "list-ol",     toolTip: `Numbered List`},
+    { iconName: "list-ul",     toolTip: `Bulleted List`},
+    { iconName: "undo",        toolTip: `Undo (${this.prefix}z)` }
+  ];
+
+  public render() {
+    const onMouseDownHandler = (event: React.MouseEvent) => {
+      event.preventDefault();
+    };
+    if (this.props.visible) {
+      const enabledClass = this.props.enabled ? "enabled" : "";
+      return (
+        <div className={`text-style-bar ${enabledClass}`}
+             onMouseDown={onMouseDownHandler}>
+          {this.renderHeaderIcon()}
+          {this.buttonDefs.map(bDef => this.renderButton(bDef))}
+        </div>
+      );
+    }
+    return (null);
+  }
+
+  private renderHeaderIcon() {
```

Comment:
While nested I think it might be useful to, whether now or later on, wrap all of `renderHeaderIcon()` in a separate `div`, then sticky that `div` of the text styling icons inside of the `text-style-bar` div so that it'll remain on the screen with their text wherever their cursor selection is (with the limitation being at the bottom of the length of the text tile).

Reply:
@llzes Makes a good point. The toolbar should be fixed while the content scrolls.

Reply:
This is an excellent catch. I've spent about an hour looking at how to use sticky positioning as @llzes suggested. However, there's enough conflict between the parent containers, overflow settings, and non-fixed height that it's going to take a few more hours to make it work. Sadly, time is too short to finish this up. Hate to do it, but I think we should defer this until time permits.